applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -357,6 +357,18 @@ namespace Kratos {
         KRATOS_CATCH("")
     }
 
+    void ContinuumExplicitSolverStrategy::ComputeNewRigidFaceNeighboursHistoricalData() {
+        KRATOS_TRY
+
+        BaseType::ComputeNewRigidFaceNeighboursHistoricalData();
+
+        IndexPartition<unsigned int>(mListOfSphericContinuumParticles.size()).for_each([&](unsigned int i){ 
+            mListOfSphericContinuumParticles[i]->ReorderFEMneighbours(); 
+        });        
```

Comment:
If `block_for_each` can be used I would use it, it is the simplest version of the parallel loop.
Only if you need the index you need to use the `IndexPartition`

In your case you don't need it:
```suggestion
        block_for_each(mListOfSphericContinuumParticles, [](typeOfVectorContentWhichIDontKnow& rParticle){ 
            rParticle.ReorderFEMneighbours(); 
        });        
```

BTW with C++14 we could replace `typeOfVectorContentWhichIDontKnow` with `auto` :/